scrypto-derive/src/blueprint.rs

```diff
@@ -278,23 +243,28 @@ fn generate_dispatcher(bp_ident: &Ident, items: &[ImplItem]) -> Result<(Vec<Expr
                     trace!("Generated stmt: {}", quote! { #stmt });
                     stmts.push(stmt);
                 }
+                stmts.push(Stmt::Expr(parse_quote! { rtn }));
 
-                arm_guards.push(parse_quote! { #fn_name });
-                arm_bodies.push(Expr::Block(ExprBlock {
-                    attrs: vec![],
-                    label: None,
-                    block: Block {
-                        brace_token: Brace {
-                            span: Span::call_site(),
-                        },
-                        stmts,
-                    },
-                }));
+                let function_ident = format_ident!("{}_{}_main", bp_ident, ident);
```

Comment:
Should we get rid of the `_main` suffix?

Reply:
That's a good idea